src/schnetpack/md/simulator.py

```diff
@@ -0,0 +1,261 @@
+        integrator (schnetpack.md.Integrator): Integrator for propagating the molecular
+                             dynamics simulation, defined in
+                             schnetpack.md.integrators
+        calculator (schnetpack.md.calculator): Calculator class used to compute molecular
+                             forces for propagation and (if requested)
+                             various other properties.
+        simulator_hooks (list(object)): List of different hooks to be applied
+                                        during simulations. Examples would be
+                                        file loggers and thermostat.
+        step (int): Index of the initial simulation step.
+        restart (bool): Indicates, whether the simulation is restarted. E.g. if set to True, the simulator tries to
+                        continue logging in the previously created dataset. (default=False)
+                        This is set automatically by the restart_simulation function. Enabling it without the function
+                        currently only makes sense if independent simulations should be written to the same file.
+        progress (bool): show progress bar during simulation. Can be deactivated e.g. for cluster runs.
+    """
+
+    def __init__(
+        self,
+        system: System,
+        integrator,
+        calculator,
+        simulator_hooks: list = [],
+        step: int = 0,
+        restart: bool = False,
+        gradients_required: bool = False,
+        progress: bool = True,
+    ):
+        super(Simulator, self).__init__()
+
+        self.system = system
+        self.integrator = integrator
+        self.calculator = calculator
+        self.simulator_hooks = torch.nn.ModuleList(simulator_hooks)
+        self.step = step
+        self.n_steps = None
+        self.restart = restart
+        self.gradients_required = gradients_required
+        self.progress = progress
+
+        # Keep track of the actual simulation steps performed with simulate calls
+        self.effective_steps = 0
+
+    @property
+    def device(self):
+        return self.system.device
+
+    @property
+    def dtype(self):
+        return self.system.dtype
+
+    def simulate(self, n_steps: int):
+        """
+        Main simulation function. Propagates the system for a certain number
+        of steps.
+
+        Args:
+            n_steps (int): Number of simulation steps to be performed.
+        """
+
+        self.n_steps = n_steps
+
+        # Determine iterator
+        if self.progress:
+            iterator = trange
+        else:
+            iterator = range
+
+        # Check, if computational graph should be built
+        if self.gradients_required:
+            grad_context = torch.no_grad()
+        else:
+            grad_context = nullcontext()
+
+        with grad_context:
+            # Perform initial computation of forces
+            self.calculator.calculate(self.system)
```

Comment:
`self.system` could turn out to be a problem...

Reply:
Keep as is for now